engine.py

```diff
@@ -152,11 +152,12 @@ def pre_app_init(self):
         except Exception as api_import_error:
             error_msg = str(api_import_error)
             self.logger.critical(error_msg)
-            QtGui.QMessageBox.critical(
-                self.get_parent_window(),
-                "Failed to import the Alias Python API",
-                error_msg,
-            )
+            if self.has_ui:
+                QtGui.QMessageBox.critical(
+                    self.get_parent_window(),
+                    "Failed to import the Alias Python API",
+                    error_msg,
+                )
```

Comment:
hey @staceyoue , we already have such a method in the engine class: https://github.com/shotgunsoftware/tk-alias/blob/master/engine.py#L641-L653
(documentation ref: https://developer.shotgridsoftware.com/tk-core/platform.html#sgtk.platform.Engine._emit_log_message)

But I'm not sure we want to display a pop-up as soon as we hit a critical log (I'm thinking of the publish logs for example)

Reply:
Yeah, I was thinking a separate function than this one, since as you mentioned it would pop up a dialog in undesirable cases. The purposed of another functions would purely be convenience to not have to remember to check the self.has_ui, but its not necessasry